include/libtorrent/aux_/disable_warnings_push.hpp

```diff
@@ -86,5 +86,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifdef _MSC_VER
 #pragma warning(push, 1)
 // warning C4005: macro redefinition
-#pragma warning( disable : 4005 )
+#pragma warning(disable : 4005)
+// expression before comma has no effect; expected expression with side-effect
+#pragma warning(disable : 4548)
```

Comment:
if you add `/Wall` to msvc you will get a lot interesting messages :)
This helps with a boost/asio import in `socket.hpp`